src/core/EdFi.DataManagementService.Core/Middleware/ValidateEqualityConstraintMiddleware.cs

```diff
@@ -22,8 +23,13 @@ internal class ValidateEqualityConstraintMiddleware(ILogger _logger, IEqualityCo
     public async Task Execute(PipelineContext context, Func<Task> next)
     {
         _logger.LogDebug("Entering ValidateEqualityConstraintMiddleware- {TraceId}", context.FrontendRequest.TraceId);
+        JsonNode? body = new JsonObject();
+        if (context.FrontendRequest.Body != null)
+        {
+            body = JsonNode.Parse(context.FrontendRequest.Body);
```

Comment:
Instead of doing this everywhere, create a new core middleware called ParseBodyMiddleware that does `JsonNode.Parse(context.FrontendRequest.Body)` on POST or PUT and stores it in a new `JsonNode?` field on `PipelineContext` called `ParsedBody`.  Then parse error handling will be in one place, and everywhere that uses `context.FrontendRequest.Body` can instead use `context.ParsedBody`.

See https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/blob/main/Meadowlark-js/packages/meadowlark-core/src/middleware/ParseBodyMiddleware.ts for an example from Meadowlark.js